src/components/filter/Filter.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useState } from 'react';
+import { StyleSheet, TouchableOpacity } from 'react-native';
+import Collapsible from 'react-native-collapsible';
+import { Divider } from 'react-native-elements';
+
+import { Icon, consts, device, normalize, texts } from '../../config';
+
+import { Button } from './../Button';
+import { BoldText } from './../Text';
+import { WrapperVertical } from './../Wrapper';
+import { DateFilter } from './DateFilter';
+import { DropdownFilter } from './DropdownFilter';
+import { StatusFilter } from './Sue';
+
+const { FILTER_TYPES, a11yLabel } = consts;
+
+type FilterProps = {
+  endDate: string;
+  serviceCode: string;
+  sort: string;
+  startDate: string;
+  status: string;
+};
+
+type FilterTypesProps = {
+  data:
+    | { id: number; index: number; selected: boolean; value: string }[]
+    | { name: string; placeholder: string }[]
+    | {
+        status: string;
+        matchingStatuses: string[];
+        iconName: string;
+      }[];
+  label?: string;
+  name: string;
+  placeholder?: string;
+  type: string;
+  value?: string;
+  startDate?: {
+    name: string;
+    placeholder: string;
+  };
+  endDate?: {
+    name: string;
+    placeholder: string;
+  };
+};
+
+type Props = {
+  filterTypes?: FilterTypesProps[];
+  setQueryVariables: (data: FilterProps) => void;
+};
+
+export const Filter = ({ filterTypes, setQueryVariables }: Props) => {
+  const [filters, setFilters] = useState({});
+  const [isCollapsed, setIsCollapsed] = useState(true);
+
+  useEffect(() => setQueryVariables(filters), [filters]);
```

Comment:
we should use `setQueryVariables` directly instead of setting filters that will set query variables?!